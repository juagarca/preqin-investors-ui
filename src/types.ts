interface IInvestor {
  firm_id: number;
  firm_name: string;
  AUM: number;
  date_added: string;
  last_updated: string;
  established_at: string;
  firm_type: string;
  city: string;
  country: string;
  address: string;
  postal_code: string;
}

export type { IInvestor };
