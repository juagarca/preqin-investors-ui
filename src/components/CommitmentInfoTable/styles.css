.commitment-info-table {
  text-align: left;
  width: fit-content;
  border: 1px solid black;
  border-collapse: collapse;
}

.commitment-info-table th,
.commitment-info-table td {
  padding: 1rem;
  border: 1px solid black;
}

.commitment-info-table thead {
  color: white;
  background: rgb(34, 93, 113);
}

.commitment-info-table tr:nth-child(even) {
  background: #f2f2f2;
}
