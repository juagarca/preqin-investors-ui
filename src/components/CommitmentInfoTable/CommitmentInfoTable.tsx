import { ICommitmentInfo } from "../../types";

import "./styles.css";

interface CommitmentInfoTableProps {
  commitmentInfoList: ICommitmentInfo[];
}

const CommitmentInfoTable = ({
  commitmentInfoList,
}: CommitmentInfoTableProps) => {
  return (
    <table className="commitment-info-table">
      <thead>
        <tr>
          <th>ID</th>
          <th>Currency</th>
          <th>Amount</th>
        </tr>
      </thead>
      <tbody>
        {commitmentInfoList.map((commitmentInfo) => (
          <TableRow key={commitmentInfo.id} commitmentInfo={commitmentInfo} />
        ))}
      </tbody>
    </table>
  );
};

const TableRow = ({ commitmentInfo }: { commitmentInfo: ICommitmentInfo }) => {
  return (
    <tr>
      <td>{commitmentInfo.id}</td>
      <td>{commitmentInfo.currency}</td>
      <td>{commitmentInfo.amount}</td>
    </tr>
  );
};

export default CommitmentInfoTable;
