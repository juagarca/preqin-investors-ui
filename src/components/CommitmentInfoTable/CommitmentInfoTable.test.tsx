import React from "react";
import { render, screen } from "@testing-library/react";

import CommitmentInfoTable from "./CommitmentInfoTable";

import { ICommitmentInfo } from "../../types";
import commitmentInfoList from "../../utils/tests/mocks/commitmentInfoList.json";

describe("CommitmentInfoTable", () => {
  const renderComponent = () =>
    render(
      <CommitmentInfoTable
        commitmentInfoList={commitmentInfoList as ICommitmentInfo[]}
      />
    );

  it("matches the snapshot", () => {
    const { asFragment } = renderComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders table rows with correct data", () => {
    renderComponent();

    expect(screen.getByText("ID")).toBeInTheDocument();
    expect(screen.getByText("Currency")).toBeInTheDocument();
    expect(screen.getByText("Amount")).toBeInTheDocument();

    commitmentInfoList.forEach((commitmentInfo) => {
      expect(screen.getByText(commitmentInfo.id)).toBeInTheDocument();
      expect(screen.getByText(commitmentInfo.currency)).toBeInTheDocument();
      expect(screen.getByText(commitmentInfo.amount)).toBeInTheDocument();
    });
  });
});
