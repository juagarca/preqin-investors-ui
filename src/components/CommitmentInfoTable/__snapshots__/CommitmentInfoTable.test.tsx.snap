// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitmentInfoTable matches the snapshot 1`] = `
<DocumentFragment>
  <table
    class="commitment-info-table"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Currency
        </th>
        <th>
          Amount
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          34728
        </td>
        <td>
          HKD
        </td>
        <td>
          6M
        </td>
      </tr>
      <tr>
        <td>
          90806
        </td>
        <td>
          EUR
        </td>
        <td>
          20M
        </td>
      </tr>
      <tr>
        <td>
          90682
        </td>
        <td>
          GBP
        </td>
        <td>
          58M
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
